openmc/mesh.py

```diff
@@ -0,0 +1,560 @@
+            Rectangular lattice used as a template for this mesh
+        division : int
+            Number of mesh cells per lattice cell.
+            If not specified, there will be 1 mesh cell per lattice cell.
+        mesh_id : int
+            Unique identifier for the mesh
+        name : str
+            Name of the mesh
+
+        Returns
+        -------
+        openmc.RegularMesh
+            RegularMesh instance
+
+        """
+        cv.check_type('rectangular lattice', lattice, openmc.RectLattice)
+
+        shape = np.array(lattice.shape)
+        width = lattice.pitch*shape
+
+        mesh = cls(mesh_id, name)
+        mesh.lower_left = lattice.lower_left
+        mesh.upper_right = lattice.lower_left + width
+        mesh.dimension = shape*division
+
+        return mesh
+
+    def to_xml_element(self):
+        """Return XML representation of the mesh
+
+        Returns
+        -------
+        element : xml.etree.ElementTree.Element
+            XML element containing mesh data
+
+        """
+
+        element = ET.Element("mesh")
+        element.set("id", str(self._id))
+
+        if self._dimension is not None:
+            subelement = ET.SubElement(element, "dimension")
+            subelement.text = ' '.join(map(str, self._dimension))
+
+        subelement = ET.SubElement(element, "lower_left")
+        subelement.text = ' '.join(map(str, self._lower_left))
+
+        if self._upper_right is not None:
+            subelement = ET.SubElement(element, "upper_right")
+            subelement.text = ' '.join(map(str, self._upper_right))
+
+        if self._width is not None:
+            subelement = ET.SubElement(element, "width")
+            subelement.text = ' '.join(map(str, self._width))
+
+        return element
+
+    @classmethod
+    def from_xml_element(cls, elem):
+        """Generate mesh from an XML element
+
+        Parameters
+        ----------
+        elem : xml.etree.ElementTree.Element
+            XML element
+
+        Returns
+        -------
+        openmc.Mesh
+            Mesh generated from XML element
+
+        """
+        mesh_id = int(get_text(elem, 'id'))
+        mesh = cls(mesh_id)
+
+        mesh_type = get_text(elem, 'type')
+        if mesh_type is not None:
+            mesh.type = mesh_type
+
+        dimension = get_text(elem, 'dimension')
+        if dimension is not None:
+            mesh.dimension = [int(x) for x in dimension.split()]
+
+        lower_left = get_text(elem, 'lower_left')
+        if lower_left is not None:
+            mesh.lower_left = [float(x) for x in lower_left.split()]
+
+        upper_right = get_text(elem, 'upper_right')
+        if upper_right is not None:
+            mesh.upper_right = [float(x) for x in upper_right.split()]
+
+        width = get_text(elem, 'width')
+        if width is not None:
+            mesh.width = [float(x) for x in width.split()]
+
+        return mesh
+
+    def build_cells(self, bc=['reflective'] * 6):
```

Comment:
The default argument for `bc` is mutable, which is a common pitfall and anti-pattern. I have provided a more detailed explanation in another PR for this issue. [Ref](https://stackoverflow.com/questions/41686829/warning-about-mutable-default-argument-in-pycharm)

Solution is the default should be `None` and in the code check

```python
if bc == None:
    bc = ['reflective'] * 6
```